spec/features/customers/show_spec.rb

```diff
@@ -0,0 +1,78 @@
+require "rails_helper"
+
+RSpec.describe "Customer's show page" do
+  before(:each) do
+    @supermarket_1 = Supermarket.create!(name: 'Corner Market', location: '123 Food St.')
+    @supermarket_2 = Supermarket.create!(name: 'HEB', location: '333  Fresh Hwy.')
+
+    @item_1 = @supermarket_1.items.create!(name: 'Syrup', price: 3)
+    @item_2 = @supermarket_1.items.create!(name: 'Bread', price: 7)
+    @item_3 = @supermarket_2.items.create!(name: 'Dough', price: 4)
+    @item_4 = @supermarket_1.items.create!(name: 'Suger', price: 10)
+
+    @customer_1 = Customer.create!(name: 'Sally Shopper')
+    @customer_2 = Customer.create!(name: "Bob Baker")
+    @customer_3 = Customer.create!(name: "Cheif Charlie ")
+
+    CustomerItem.create(customer: @customer_1, item: @item_1)
+    CustomerItem.create(customer: @customer_1, item: @item_2)
+    CustomerItem.create(customer: @customer_2, item: @item_3)
+    
+  end
+
+      #   Story 1
+      # As a visitor, 
+      # When I visit a customer show page,
+      # I see the customer's name,
+      # And I see its a list of its items
+      # including the item's name, price, and the name of the supermarket that it belongs to.
+  describe "When I visit a customer show page" do
+    it "I see the customer's name and And I see its a list of its items" do
+      visit "/customers/#{@customer_1.id}"
+
+      expect(page).to have_content(@customer_1.name)
+      expect(page).to have_content(@item_1.name)
+      expect(page).to have_content(@item_1.price)
+      expect(page).to have_content(@supermarket_1.name)
+      expect(page).to have_content(@item_2.name)
+      expect(page).to have_content(@item_2.price)
+      expect(page).to have_content(@supermarket_1.name)
```

Comment:
These assertions would be more meaningful with `within` blocks! For example, `expect(page).to have_content(@item_2.price)` just resolves to expecting a `7` somewhere on that page. That's not super helpful if there are other 7s on the page from other prices, or maybe phone numbers and addresses.

Your view is also already set up with text labels so you could change these assertions to include those like so:
`      expect(page).to have_content("Item Price: #{@item_2.price}")`
In this case, the price is anchored to the text before it which helps make it more meaningful.